### ISSUE TITLE
Enum is considered as change when its fetched through query.

### ISSUE THREAD

**Issue by Warxcell** [doctrine/doctrine2#10246]
### Bug Report

|    Q        |   A
|------------ | ------
| BC Break    | no
| Version     | 2.13.3

#### Summary

I've added a test case that reproduces the bug. 

https://github.com/doctrine/orm/pull/10245

#### Current behavior

https://github.com/doctrine/orm/pull/10245

Its even worst when entity have READ_ONLY cache, because CachePersister throws error that it cannot update the entity.

#### How to reproduce

https://github.com/doctrine/orm/pull/10245

#### Expected behavior

https://github.com/doctrine/orm/pull/10245

Basically when there is no change - to not flush entity.

**Comment by Warxcell** [comment]
Resolved by https://github.com/doctrine/orm/pull/10074
